memguard.go

```diff
@@ -38,13 +39,21 @@ Similarly, if a function is given a LockedBuffer that has been
 destroyed, the call will return an ErrDestroyed.
 */
 type LockedBuffer struct {
+	*lockedBuffer
+	*finalizerGuard
+}
+
+// LockedBuffer is actually just a handle to lockedBuffer
+type lockedBuffer struct {
 	sync.Mutex
 	Buffer []byte
 
 	readOnly  bool
 	destroyed bool
 }
 
+type finalizerGuard [16]byte
```

Comment:
Going by the documentation, no. As long as you allocate it at runtime and the type size is greater than 0 it should be enough. Unfortunately it seems that when tiny allocations (<16 bytes) are used finalizers can behave weirdly and randomly fail to run (that's the reason the first version of the PR was hanging go test -race on mac/linux; I was using an int instead of [16]byte). I should probably report this as a golang bug; in the meantime we use [16]byte here so that this does not affect us.